### PR TITLE
Fix shortenlabel repo equals rule case

### PR DESCRIPTION
**Pull request by borkaehw**
# Issue
We are failing to shorten label when repo equals to rule, to be specific
`@foo//:foo` should be shortened to `@foo`
Also, we add test cases for this issue.

### CODE REVIEW

File: edit/edit_test.go [hunk 1]
```diff
@@ -62,11 +62,14 @@ var shortenLabelTests = []struct {
...
 	{"//base:base", "devtools", "//base"},
 	{"//base", "base", ":base"},
+	{"//devtools/buildozer:buildozer", "", "//devtools/buildozer"},
 	{"@r//devtools/buildozer:buildozer", "devtools/buildozer", "@r//devtools/buildozer"},
 	{"@r//devtools/buildozer", "devtools/buildozer", "@r//devtools/buildozer"},
 	{"@r//devtools", "devtools", "@r//devtools"},
 	{"@r:rule", "", "@r:rule"},
 	{"@r", "", "@r"},
+	{"@foo//:foo", "", "@foo"},
+	{"@foo//devtools:foo", "", "@foo"},
```
Comment (pmbethe09) [inline edit/edit_test.go]: this looks wrong.  while the previous should shorten, here it should remain as @foo//devtools:foo and the package should be devtools

File: edit/edit.go [hunk 2]
```diff
@@ -90,6 +90,9 @@ func ShortenLabel(label string, pkg string) string {
 		}
 		return "@" + repo + "//" + labelPkg
 	}
+	if strings.HasPrefix(label, "@") && repo == rule {
```
Comment (pmbethe09) [inline edit/edit.go]: based on your test, this looks incomplete

**Comment by borkaehw** [comment]
Thanks for pointing the problem out, hopefully this makes more sense. I also added more test cases.

File: edit/edit.go (hunk 2 above)
Comment (borkaehw) [reply edit/edit.go]: I've made some changes.

Comment (pmbethe09) [reply edit/edit.go]: it seems to me this simplifies as just
if  strings.HasPrefix(label, "@") && repo == rule && labelPkg == "" {
￼   return "@" + repo
￼ }

File: edit/edit_test.go [hunk 3]
```diff
@@ -58,15 +58,21 @@ var shortenLabelTests = []struct {
...
 	{"//base:base", "devtools", "//base"},
 	{"//base", "base", ":base"},
+	{"//devtools/buildozer:buildozer", "", "//devtools/buildozer"},
 	{"@r//devtools/buildozer:buildozer", "devtools/buildozer", "@r//devtools/buildozer"},
 	{"@r//devtools/buildozer", "devtools/buildozer", "@r//devtools/buildozer"},
 	{"@r//devtools", "devtools", "@r//devtools"},
 	{"@r:rule", "", "@r:rule"},
 	{"@r", "", "@r"},
+	{"@foo//:foo", "", "@foo"},
+	{"@foo//devtools:foo", "", "@foo//devtools:foo"},
```
Comment (pmbethe09) [inline edit/edit_test.go]: this looks out of place, same LHS as the next case, but empty 'pkg' here looks wrong.

Comment (borkaehw) [reply edit/edit_test.go]: As far as I know, we are referring to external repo while we are in the current workspace root. For the next case, we are also referring to external repo but we want to make sure even though we have the same package name, we shouldn't shorten it anyway.

Could you elaborate more what looks wrong to you? Thanks.

File: edit/edit.go (hunk 2 above)
Comment (borkaehw) [reply edit/edit.go]: This makes sense, thanks for feedback again.

File: edit/edit_test.go (hunk 3 above)
Comment (pmbethe09) [reply edit/edit_test.go]: only one of
{"@foo//devtools:foo", "", "@foo//devtools:foo"},
{"@foo//devtools:foo", "devtools", "@foo//devtools:foo"},

should work, as with the same input you expect the same output -- these are 2 tests with the same input but 2 different outputs.

**Comment by borkaehw** [comment]
@pmbethe09 I would like to know if there is other improvement we should make to get this merged?

**Comment by pmbethe09** [comment]
LGTM (i was away).  Also hecking with @vladmos

**Pull request merged**